specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+deletion.
+
+### Channel Lifecycle
+
+A Channel represents an Addressable endpoint (i.e. it has as
+`status.address.url` field) which can receive, store, and forward events to
+multiple recipients (Subscriptions). Subscriptions are associated with a Channel
+based on the `spec.channel` field on the Subscription; it is expected that the
+controller for a Channel will also control the associated Subscriptions. When
+the Channel's `Ready` condition is `true`, the Channel MUST provide a
+`status.address.url` which accepts all valid CloudEvents and MUST forward the
+received events to each associated Subscription whose `Ready` condition is
+`true`. As described in the [Subscription Lifecycle](#subscription-lifecycle)
+section, a Channel MAY forward events to an associated Subscription which does
+not currently have a `true` `Ready` condition, including events received by the
+Channel before the `Subscription` was created.
+
+When a Channel is created, its `spec.channelTemplate` field MUST be populated to
+indicate which of several possible Channel implementations to use. It is
+RECOMMENDED to default the `spec.channelTemplate` field on creation if it is
+unpopulated. Once created, the `spec.channelTemplate` field MUST be immutable;
+the Channel MUST be deleted and re-created to change the `spec.channelTemplate`.
+This pattern is chosen to make it clear that changing `spec.channelTemplate` is
+not an atomic operation and that any implementation would be likely to result in
+event loss during the transition.
+
+### Subscription Lifecycle
+
+The lifecycle of a Subscription is independent of that of the channel it refers
+to in its `spec.channel` field. The `spec.channel` object reference may refer to
+either an `messaging.knative.dev/v1` Channel resource, or another resource which
+meets the `spec.subscribers` and `spec.delivery` required elements in the
+Channelable duck type. If the referenced `spec.channel` does not currently exist
+or its `Ready` condition is not `true`, then the Subscription's `Ready`
+condition MUST NOT be `true`, and the reason SHOULD indicate that the
+corresponding channel is missing or not ready.
+
+The Subscription MUST also set the `status.physicalSubscription` URIs by
+resolving the `spec.subscriber`, `spec.reply`, and
+`spec.delivery.deadLetterSink` as described in
+[Addressable resolution](#addressable-resolution) before setting the `Ready`
+condition to `true`. If any of the addressable fields fails resolution, the
+Subscription MUST set the `Ready` condition to `false`, and at least one
+condition should indicate the reason for the error. (It is acceptable for none
+of the `spec.subscriber`, `spec.reply`, and `spec.delivery.deadLetterSink`
+fields to contain a `ref` field.)
+
+At least one of `spec.subscriber` and `spec.reply` MUST be set; if only
+`spec.reply` is set, the behavior is equivalent to setting `spec.subscriber`
+except that the Channel SHOULD NOT advertise the ability to process replies
```

Comment:
should be: if only **`spec.subscriber`** is set ...

Reply:
unless the intent is to say that when `subscriber` is not set and `reply` is set then the Channel SHOULD NOT ....

Reply:
Yes, the intent is to say that setting `subscriber` but not `reply` should be distinguishable by the presence of the `Prefer: Reply` header.

I have a PR currently in testing to implement this for `knative/eventing`, and then the other sandbox implementations will come next.